### DISCUSSION TITLE
Support for 3D lookup table?

### DISCUSSION THREAD

**Discussion by yuxng** [casadi/casadi#3648] [Q&A] [answered]
Hello!
From the casadi document, I see that we can do 1D lookup tables and 2D lookup tables using the interpolant function.
Is it possible to build 3D lookup tables?

My problem is that, I have a 3D signed distance field that I would like to use as a cost function in the optimization.
A 3D lookup table seems needed in this case. Any suggestion is appreciated.

Thanks!

**Comment by b-mare** [comment, marked as answer]
Hi @yuxng , 

The documentation mentions it is possible for an arbitrary number of dimensions.
[https://web.casadi.org/docs/#using-lookup-tables](url) 

Example
```
import numpy as np
import casadi as ca
from scipy.interpolate import RegularGridInterpolator

xgrid = np.linspace(0, 4, 3)
ygrid = np.linspace(10, 16, 4)
zgrid = np.linspace(20, 30, 5)
X, Y, Z = np.meshgrid(xgrid, ygrid, zgrid, indexing="ij")
data = 2 * X + Y / Z
data_flat = data.ravel(order="F")
lut = ca.interpolant("lut", "linear", [xgrid, ygrid, zgrid], data_flat)
interp_func = RegularGridInterpolator((xgrid, ygrid, zgrid), data, method='linear')

print(lut([2, 12, 27.5]))
print(interp_func([2, 12, 27.5]))
```

> 4.43636
[4.43636364]

**Reply by yuxng** [reply]
Oh, I will try this. Thanks a lot!
